language-design.md

```diff
@@ -38,7 +53,11 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+In Growing a Language, Steele emphasizes the importance of extensibility in language design. In our lab, we explored actually implementing some of that extensibility. One idea presented in Growing a Language was definitely evident in our lab--the slowness of trying to grow a language all by oneself [Steele 1998, 6]. Almost everyone in the room did not feel they were able to make much progress, in part because of the difficulty of working by oneself. However, if we were to combine the progress of everyone in the room, we would have quickly expanded the music library by a quite greater magnitude.
+
+Also, the “bazaar-vs-cathedral” idea mentioned in our response to the quote question is also relevant here. Given that the music library was much more a library than a language, it definitely had the open feel of a bazaar, without much structure to guide how it _should_ work. It felt like we had the task of creating the shopping mall structure ourselves, rather than working within an existing shopping mall.
 
+Essentially, the music library was missing that which Steele considers key: “a pattern for growth” [Steele 1998, 12]. It did not have a basic structure for development and extension. However, because it was bare bones enough, that means that we as developers could add that pattern in as we extend and modify it. We think that’s what many of us intuitively tried to add to the language as we extended it--patterns, systems, etc. 
```

Comment:
This is a very different take on this question than we had - I definitely agree with your train of thought here. The idea that everyone working together would make more progress makes sense and fits in with the paper, as does the bazaar/cathedral idea. The idea that we were designing the language or library's pattern for growth is interesting because it seems like it gives us more powers than typical users of a language would have in design - it seems like it really ought to be the responsibility of the language designer to provide that pattern for growth, since you are viewing all of us as users of the language. I guess we can blame Prof. Dodds for that.

The thing that our answers have in common here is the idea that we were adding features that were generally extensible. I think this is a good way of looking at why we all intuitively tried to do that.